### DISCUSSION TITLE
Can I use some of the model events for custom logging?

### DISCUSSION THREAD

**Discussion by seabasss** [spatie/laravel-activitylog#1048] [General]
I like to log things right when it happens in the controller code. I see that there are a lot of nice methods for models, but are there similar methods for custom logging? Especially `logOnlyDirty()`?

Thank you!

**Comment by Gummibeer** [comment]
I'm on phone right now - so no explicit code. But there's one method in the traits used to get the changes. I think it's not public so you could add a public `getChangesForLog` method to you model making that logic available to the outside.

Keep in mind that you will have to call it after filling data but before persisting to database.
